dd-sdk-android-gradle-plugin/src/main/kotlin/com/datadog/gradle/plugin/internal/sanitizer/DdUrlSanitizer.kt

```diff
@@ -0,0 +1,41 @@
+package com.datadog.gradle.plugin.internal.sanitizer
+
+import java.util.Locale
+
+/**
+ * It will strip out the user information from the provided URL by applying the following rules:
+ * 1. In case the URL is a valid schema URL : ((ssh|http[s]?):\\) it will strip out the user
+ * information if any from it
+ * 2. In case the URL is a valid schema URL but by mistake the user forgot to add the schema:
+ * [username]:[password]@github.com/(.+).git it will strip out the user information if any and will
+ * return the new url
+ * 3. In case the URL is of a GIT SSH short format: [user@]github.com:(.+).git it will do
+ * nothing assuming that there is no password normally provided in a GIT SSH short format URL.
+ */
+internal class DdUrlSanitizer(
+    private val strategyResolver: (String) -> SanitizerStrategy = {
+        if (it.matches(Regex(VALID_SCHEMA_URL_FORMAT_REG_EX, RegexOption.IGNORE_CASE))) {
+            DefaultSanitizerStrategy()
+        } else {
+            NoSchemaSanitizerStrategy()
+        }
+    }
+) : UrlSanitizer {
+
+    @SuppressWarnings("TooGenericExceptionCaught")
+    override fun sanitize(url: String): String {
+        try {
+            return strategyResolver(url).apply(url)
+        } catch (e: Exception) {
+            throw UriParsingException(WRONG_URL_FORMAT_ERROR_MESSAGE.format(url, Locale.US), e)
+        }
+    }
+
+    companion object {
+        const val WRONG_URL_FORMAT_ERROR_MESSAGE =
+            "The detected GIT remote url: [%s] is not a valid GIT url and it " +
+                "will not be browsable when using it from error dashboard. " +
+                "You can provide a different URL through the plugin extension."
+        const val VALID_SCHEMA_URL_FORMAT_REG_EX = "^(ssh|http|https)://(.+)$"
```

Comment:
I'd rename that `…_PATTERN` and create a static `…_REGEX = Regex(VALID_SCHEMA_URL_FORMAT_PATTERN)`